plugins/base/proto/base.proto

```diff
@@ -87,6 +87,40 @@ message NomadDriverConfig {
     // communicating with plugin subsystems over loopback
     // buf:lint:ignore FIELD_LOWER_SNAKE_CASE
     uint32 ClientMinPort = 2;
+
+    // Topology is the complex hardware topology detected by the client
+    // combined with client configuration.
+    ClientTopology Topology = 3;
```

Comment:
Doesn't have to be in this PR, but there's documentation for the [`SetConfig`](https://developer.hashicorp.com/nomad/docs/concepts/plugins/base#setconfig-config-config-error) RPC to update with this.